src/cca/scripts/background.ts

```diff
@@ -0,0 +1,61 @@
+/// <reference path='../../../third_party/cordova/splashscreen.d.ts'/>
+/// <reference types="chrome/chrome-app" />
+
+/**
+ * background.ts
+ *
+ * This file must be included *after* the freedom script and manifest are
+ * loaded.
+ */
+
+import { ServerListPage } from '../ui_components/server_list';
+import { UproxyServerRepository } from './uproxy_server';
+import { MakeCoreConnector } from './cordova_core_connector';
+import { GetGlobalTun2SocksVpnDevice } from './tun2socks_vpn_device';
+import * as vpn_device from '../model/vpn_device';
+import * as intents from './intents';
+
+// We save this reference to allow inspection of the context state from the browser debuggin tools.
+(window as any).context = this;
+
+// TODO(fortuna): Get rid of core connector and talk to the core directly.
+// TODO(fortuna): I believe we need to somehow wait for the core to be ready.
+let core = MakeCoreConnector();
+
+// Create UproxyServerRepository.
+let serversPromise = GetGlobalTun2SocksVpnDevice().then((vpnDevice) => {
+  console.debug('Device supports VPN');
+  return vpnDevice;
+}).catch((error) => {
+  console.error(error);
+  return new vpn_device.NoOpVpnDevice();
+}).then((vpnDevice) => {
+  return new UproxyServerRepository(core, vpnDevice);
+});
+
+// Create UI.
+let serversListPagePromise: Promise<ServerListPage> = new Promise((resolve, reject) => {
```

Comment:
The Google style guides usually say you should avoid abbreviations ([example](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules)). With the long names, the meaning is clear. In fact, when I first came to this code, it wasn't clear to me what `F` and `R` where. My first guess was Fail and Resolve, which was wrong and I had to look at the documentation to check what they actually were.

For that reason, I'd favor making the code clearer and more explicit over making it look good. I'm happy to change to something else that is clear. In other places of the code I've seen `fulfill` and `reject`. But I can do `F` and `R` if you feel strongly about it.